eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -174,7 +174,7 @@ import Cardano.Ledger.Block (Block)
 import Cardano.Ledger.Coin
 import Cardano.Ledger.Compactible (fromCompact)
 import Cardano.Ledger.Credential (Credential (..), Ptr, StakeReference (..), credToText)
-import Cardano.Ledger.Genesis (EraGenesis (..), NoGenesis (..))
+import qualified Cardano.Ledger.Genesis as Genesis
```

Comment:
That's fun, I forgot that there is a `Genesis` key role. This guys are no longer exist starting with Conway era. Instead of having conflicting definitions for totally unrelated concepts with the same name let's instead rename to `GenesisRole` in here: https://github.com/IntersectMBO/cardano-ledger/blob/d120d9613810c0dc9e22fe2d1c104d8240aeb163/libs/cardano-ledger-core/src/Cardano/Ledger/Keys/Internal.hs#L68
```suggestion
import Cardano.Ledger.Genesis
```

Feel free to rename it without any deprecation procedures, just make sure to add this rename to the changelog.